detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnIfStatements.kt

```diff
@@ -0,0 +1,178 @@
+        var current: KtExpression? = expression
+        while (current is KtIfExpression) {
+            current.then?.let { list.add(it) }
+            // Don't add `if` because it's an `else if` which we treat as one unit.
+            current.`else`?.takeIf { it !is KtIfExpression }?.let { list.add(it) }
+            current = current.`else`
+        }
+        return list
+    }
+
+    private fun validate(list: List<KtExpression>, policy: BracePolicy) {
+        when (policy) {
+            BracePolicy.Always -> {
+                if (!list.all { hasBraces(it) }) {
+                    val violator = list.first { !hasBraces(it) }
+                    report(violator, policy)
+                }
+            }
+
+            BracePolicy.Never -> {
+                if (!list.all { !hasBraces(it) }) {
+                    val violator = list.first { hasBraces(it) }
+                    report(violator, policy)
+                }
+            }
+
+            BracePolicy.Consistent -> {
+                val reference = hasBraces(list.first())
+                if (!list.all { hasBraces(it) == reference }) {
+                    val violator = list.first { hasBraces(it) != reference }
+                    report(violator, policy)
+                }
+            }
+        }
+    }
+
+    private fun report(violator: KtExpression, policy: BracePolicy) {
+        val iff = violator.parent.parent as KtIfExpression
+        val reported = when {
+            iff.then === violator -> iff.ifKeyword
+            iff.`else` === violator && isMultiStatement(violator) -> iff.ifKeyword
+            iff.`else` === violator -> iff.elseKeyword
+            else -> error("Violating element ($violator) is not part of this if: $iff")
+        }
+        report(CodeSmell(issue, Entity.from(reported ?: violator), issue.description + ": $policy"))
+    }
+
+    private fun isMultiStatement(expression: KtExpression) =
+        expression is KtBlockExpression && expression.statements.size > 1
+
+    private fun policy(expression: KtExpression): BracePolicy =
+        if (expression.textContains('\n')) multiLine else singleLine
+
+    private fun hasBraces(expression: KtExpression): Boolean = expression is KtBlockExpression
+
+    companion object {
+        private const val BRACES_ALWAYS = "always"
+        private const val BRACES_NEVER = "never"
+        private const val BRACES_CONSISTENT = "consistent"
+    }
+
+    enum class BracePolicy {
+        Always,
+        Never,
+        Consistent;
+
+        companion object {
+            fun getValue(arg: String): BracePolicy =
+                when (arg) {
+                    BRACES_NEVER -> Never
+                    BRACES_ALWAYS -> Always
+                    BRACES_CONSISTENT -> Consistent
+                    else -> error(
+                        "Unknown value $arg, allowed values are: $BRACES_NEVER|$BRACES_ALWAYS|$BRACES_CONSISTENT"
+                    )
+                }
+        }
+    }
```

Comment:
What do you think about?
```suggestion
    enum class BracePolicy(private val config: String) {
        Always("always"),
        Never("never"),
        Consistent("consistent");

        companion object {
            fun getValue(arg: String): BracePolicy =
                values().singleOrNull { it.config == arg }
                   ?: error("Unknown value ${arg}, allowed values are: ${values().joinToString("|")}")
                }
        }
    }
```